rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -0,0 +1,413 @@
+        # (combines multiple UTXOs for the same address)
+        per_address_in_out_totals = {
+            True: defaultdict(lambda: ZERO),
+            False: defaultdict(lambda: ZERO),
+        }
+        for tx_in_out_list, is_output in (
+                (tx.inputs, False),
+                (tx.outputs, True),
+        ):
+            per_address_totals = per_address_in_out_totals[is_output]
+            for tx_io in tx_in_out_list:
+                # Handle various types of scripts
+                if tx_io.type == BtcScriptType.OP_RETURN:
+                    if (event := self.decode_op_return(tx=tx, raw_script=tx_io.raw_script)) is not None:
+                        events.append(event)
+                    else:
+                        log.error(f'Failed to decode OP_RETURN bitcoin script in {tx.tx_id} {tx_io}')
+                    continue
+                elif (
+                    (address := tx_io.address) is None and
+                    tx_io.type == BtcScriptType.P2PK
+                ):  # derive the public key hash address from the raw public key used by p2pk
+                    address = derive_p2pkh_from_p2pk(tx_io.pubkey)
+                elif address is None:
+                    log.error(f'Encountered {tx_io.type} script with no address in transaction {tx.tx_id}')
+                    continue
+
+                per_address_totals[address] += tx_io.value
+
+        # Create fee events
+        total_input = sum(per_address_in_out_totals[False].values())
+        for address, amount in per_address_totals.items():
+            per_address_totals[address] -= (fee_share := (amount / total_input) * tx.fee)
+            if address not in self.tracked_accounts:
+                continue  # only add the fee event for tracked accounts
+
+            events.append(self.create_event(
+                tx=tx,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.FEE,
+                amount=fee_share,
+                notes=f'Burn {fee_share} BTC for gas',
+                location_label=address,
+            ))
+
+        input_totals = {k: v for k, v in per_address_in_out_totals[False].items() if v != ZERO}
+        output_totals = {k: v for k, v in per_address_in_out_totals[True].items() if v != ZERO}
+        in_len, out_len = len(input_totals), len(output_totals)
+        if in_len == 0 or out_len == 0:
+            # Inputs have all been spent on a fee or outputs have no value (likely op_return)
+            if in_len != out_len:
+                # only one is zero length - shouldn't happen since there would have to be
+                # either more btc going in than came out, or vice versa
+                log.error(
+                    'Encountered Bitcoin transaction with mismatched '
+                    'input/output amounts. Should not happen.',
+                )
+
+            return events  # no need to process further
```

Comment:
```suggestion
        if in_len == 0 or out_len == 0 or in_len != out_len:
                # Inputs have all been spent on a fee or outputs have no value (likely op_return)
                # only one is zero length - shouldn't happen since there would have to be
                # either more btc going in than came out, or vice versa
                log.error(
                    'Encountered Bitcoin transaction with mismatched '
                    'input/output amounts. Should not happen.',
                )

            return events  # no need to process further
```

you'll need to adjust the logs and comments